src/util/fetch.ts

```diff
@@ -172,13 +196,13 @@ export default function fetchFile(options) {
             }
 
             // fire progress event when during load
-            instance.onProgress = e => {
+            instance.onProgress = (e: ProgressEvent) => {
                 instance.fireEvent('progress', e);
             };
 
             return new Response(
                 new ReadableStream(
-                    new ProgressHandler(instance, contentLength, response)
+                    new ProgressHandler(instance, contentLength!, response)
```

Comment:
??

Reply:
Asserts not null :D 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Code above already checks `if (contentLength === null)` but typescript didn't understand that and still had a warning without the !

Reply:
hmm i sure hope there are not many more of these weird checks necessary or it'll be really hard to contribute in a fun and easy way in the future.. i didn't expect the complete codebase to be converted into typescript..

Reply:
I think by rewriting the code in that function the `!` wouldn't be necessary. I think setting `progressAvailable = false` instead of returning is preventing typescript from understanding that contentLength cannot be null.

But I didn't want to rewrite any code in this PR unless necessary, so I don't accidentally change how the code works.